orb_models/forcefield/conservative_regressor.py

```diff
@@ -237,25 +254,38 @@ def loss(self, batch: base.AtomGraphs) -> base.ModelOutput:
         total_loss += loss
         metrics.update({f"{self.grad_prefix}-{k}": v for k, v in loss_out.log.items()})
 
-        # Conservative stress
-        loss_out = stress_loss_function(
-            pred=grad_stress_pred,
-            raw_target=batch.system_targets[self.stress_name],
-            raw_gold_target=batch.system_targets[self.stress_name],
-            name=self.stress_name,
-            normalizer=self.grad_stress_normalizer,
-            loss_type=energy_head.loss_type,
-        )
-        loss = self.loss_weights.get(self.grad_stress_name, 1.0) * loss_out.loss
-        loss_out.log[f"{self.grad_stress_name}_loss"] = loss
-        total_loss += loss
-        metrics.update({f"{self.grad_prefix}-{k}": v for k, v in loss_out.log.items()})
+        # Conservative stress (optional)
+        if self.has_stress and self.grad_stress_name in out:
+            assert self.stress_name is not None
+            assert self.grad_stress_name is not None
+            raw_grad_stress_pred = out[self.grad_stress_name]
+            grad_stress_pred = self.grad_stress_normalizer(
+                raw_grad_stress_pred, online=False
+            )
+            loss_out = stress_loss_function(
+                pred=grad_stress_pred,
+                raw_target=batch.system_targets[self.stress_name],
+                raw_gold_target=batch.system_targets[self.stress_name],
+                name=self.stress_name,
+                normalizer=self.grad_stress_normalizer,
+                loss_type=energy_head.loss_type,
+            )
+            loss = self.loss_weights[self.grad_stress_name] * loss_out.loss
```

Comment:
I think it be better to use `self.loss_weights.get(self.grad_stress_name, 1.0)`, if only for consistency with the other losses above.